<a href="https://colab.research.google.com/github/GAffig/DS4A-/blob/main/Team25_Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
#pip install  pandasql
#from pandasql import sqldf

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving mass_killing_incidents_public.csv to mass_killing_incidents_public (1).csv


In [ ]:
import io
inc = pd.read_csv(io.BytesIO(uploaded['mass_killing_incidents_public.csv']))


In [ ]:
vic = df1
vic.head()


1

In [87]:
#vic[vic['incident_id'].isna()]
#vic = vic.dropna(subset=['incident_id'])
#vic.incident_id= vic.incident_id.astype(int)

vic.head()

incident_id  victim_id   age             race     sex vorelationship  \
0            1          1  25.0  Hispanic/Latino    Male   Undetermined   
1            1          2  30.0  Hispanic/Latino    Male   Undetermined   
2            1          3  30.0  Hispanic/Latino    Male   Undetermined   
3            1          4  32.0  Hispanic/Latino    Male   Undetermined   
4            2          5  30.0            White  Female       Neighbor   

   incident  
0       1.0  
1       1.0  
2       1.0  
3       1.0  
4       2.0

In [ ]:
vic.drop()

/usr/local/lib/python3.8/dist-packages/pandas/core/generic.py:5516: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


In [80]:
inc.head()

incident_id        date              city state  num_offenders  \
371            1  2010-09-02           Chicago    IL              2   
369            2  2010-09-11           Jackson    KY              1   
370            3  2010-09-04      Lake Charles    LA              1   
367            4  2010-09-28            Boston    MA              3   
372            5  2010-08-28  Lake Havasu City    AZ              1   

     num_victims_killed  num_victims_injured  firstcod secondcod    type  \
371                   4                  0.0  Shooting       NaN  Felony   
369                   5                  0.0  Shooting       NaN  Family   
370                   4                  0.0  Shooting       NaN   Other   
367                   4                  1.0  Shooting       NaN  Felony   
372                   5                  1.0  Shooting       NaN  Family   

             situation_type            location_type   location   longitude  \
371              Drug trade  Residence/Other shelter  Residence  -87.658536   
369  Interpersonal conflict  Residence/Other shelter  Residence  -83.383656   
370  Interpersonal conflict  Residence/Other shelter  Residence  -93.177638   
367                 Robbery  Residence/Other shelter  Residence  -71.089128   
372  Interpersonal conflict  Residence/Other shelter  Residence -114.322455   

      latitude                                          narrative  
371  41.987533  Police tailing suspects in a series of drug-re...  
369  37.553424  When Stanley Neace's wife didn't cook his eggs...  
370  30.156163  Lee Roy Williams told police in a recorded int...  
367  42.282016  An $1,800 robbery ended in the shootings of fi...  
372  34.483901  Brian Diez, 26, barged into his ex-girlfriend'...

In [82]:
query = """ SELECT st.Students, st.Gender, st.Email, st.Age, tat.Department
            FROM students_df st INNER JOIN teaching_assistant_df tat 
            ON st.Email = tat.Email;
            """

result = sqldf(query)
result

371    False
369    False
370    False
367    False
372    False
       ...  
4      False
3      False
2      False
1      False
0      False
Name: incident_id, Length: 523, dtype: bool

In [93]:
query = """ SELECT *
            FROM vic INNER JOIN inc 
            ON vic.incident_id = inc.incident_id;
            """

result = sqldf(query)


In [109]:
result.head(10)
result.firstcod.unique()

array(['Shooting', 'Stabbing', 'Smoke inhalation & burns', 'Blunt force',
       'Strangulation', 'Vehicle crash', 'Pushing/Jumping', 'Drowning',
       'Unknown', 'Asphyxiation'], dtype=object)

In [105]:
result.state = result.state.astype('category')

In [107]:
result.state.value_counts()

CA    291
TX    280
FL    183
IL    145
OH    119
NY    106
VA    103
GA     81
NV     80
PA     76
WA     75
AZ     75
MI     70
NC     64
SC     63
MO     63
WI     61
IN     61
AL     57
LA     56
TN     53
CO     51
OK     47
CT     40
MD     38
OR     35
KS     35
MS     32
NJ     28
MA     28
WV     26
KY     26
DC     24
NM     19
MN     18
AR     17
NE     16
UT     13
ND     12
IA     10
AK      8
ID      8
DE      8
WY      8
SD      5
MT      4
VT      4
ME      4
Name: state, dtype: int64

## FOCUS B4 SATURDAY
* life expectancy at birth dataset by state and standard error
* grouped by sex 
* we neeed 2016 and 2017

### States
* CA    291
* TX    280
* FL    183
* IL    145
* OH    119